pkg/platformconfig/types.go

```diff
@@ -184,21 +184,24 @@ type PlatformKubeConfig struct {
-	DefaultFunctionServiceAccount    string                  `json:"defaultFunctionServiceAccount,omitempty"`
-	ValidFunctionPriorityClassNames  []string                `json:"validFunctionPriorityClassNames,omitempty"`
-	DefaultFunctionPodResources      PodResourceRequirements `json:"defaultFunctionPodResources,omitempty"`
-	DefaultSidecarResources          PodResourceRequirements `json:"defaultSidecarResources,omitempty"`
-	DefaultFunctionTolerations       []corev1.Toleration     `json:"defaultFunctionTolerations,omitempty"`
-	PreemptibleNodes                 *PreemptibleNodes       `json:"preemptibleNodes,omitempty"`
-	DefaultReadinessProbe            *corev1.Probe           `json:"readinessProbe,omitempty"`
-	DefaultLivenessProbe             *corev1.Probe           `json:"livenessProbe,omitempty"`
-	ElasticSearchConfig              *ElasticSearchConfig    `json:"elasticSearchConfig,omitempty"`
+	DefaultServiceType                     corev1.ServiceType      `json:"defaultServiceType,omitempty"`
+	DefaultFunctionNodeSelector            map[string]string       `json:"defaultFunctionNodeSelector,omitempty"`
+	DefaultHTTPIngressHostTemplate         string                  `json:"defaultHTTPIngressHostTemplate,omitempty"`
+	DefaultHTTPIngressAnnotations          map[string]string       `json:"defaultHTTPIngressAnnotations,omitempty"`
+	DefaultHTTPIngressClassName            string                  `json:"defaultHTTPIngressClassName,omitempty"`
+	DefaultFunctionPriorityClassName       string                  `json:"defaultFunctionPriorityClassName,omitempty"`
+	DefaultFunctionServiceAccount          string                  `json:"defaultFunctionServiceAccount,omitempty"`
+	ValidFunctionPriorityClassNames        []string                `json:"validFunctionPriorityClassNames,omitempty"`
+	DefaultFunctionPodResources            PodResourceRequirements `json:"defaultFunctionPodResources,omitempty"`
+	DefaultSidecarResources                PodResourceRequirements `json:"defaultSidecarResources,omitempty"`
+	DefaultFunctionTolerations             []corev1.Toleration     `json:"defaultFunctionTolerations,omitempty"`
+	PreemptibleNodes                       *PreemptibleNodes       `json:"preemptibleNodes,omitempty"`
+	DefaultReadinessProbe                  *corev1.Probe           `json:"readinessProbe,omitempty"`
+	DefaultLivenessProbe                   *corev1.Probe           `json:"livenessProbe,omitempty"`
+	ElasticSearchConfig                    *ElasticSearchConfig    `json:"elasticSearchConfig,omitempty"`
+	ProjectSecretTemplate                  string                  `json:"projectSecretTemplate,omitempty"`
+	ProjectSecretAllowedServiceAccountsKey string                  `json:"projectSecretAllowedServiceAccountsKey,omitempty"`
+	ProjectSecretDefaultServiceAccountKey  string                  `json:"projectSecretDefaultServiceAccountKey,omitempty"`
```

Comment:
Once the feature is done - add a section to `configuring-a-platform.md` (open a ticket for it)

Reply:
https://iguazio.atlassian.net/browse/NUC-521